src/background/services/network/handlers/wallet_addEthereumChain.ts

```diff
@@ -118,8 +118,14 @@ export class WalletAddEthereumChainHandler extends DAppRequestHandler {
     }
     const skipApproval = await isCoreWeb(request);
 
+    const targetNetwork = chainRequestedIsSupported
```

Comment:
Should this be handled in the `setNetwork` function of the NetworkService instead? Ensuring that we are using the exact network object from the network service appears to be quite error prone.

Reply:
I updated `NetworkService.setNetwork()` to accept the chain id instead, this guarantees the network needs to exist in the `NetworkService` first (so dapps / handlers need to first `.saveCustomNetwork()` and then try to switch to it, otherwise it won't work - previously it would).